libbpf-tools/opensnoop.c

```diff
@@ -79,7 +81,7 @@ static const struct argp_option opts[] = {
 static error_t parse_arg(int key, char *arg, struct argp_state *state)
 {
 	static int pos_args;
-	int pid, uid, duration;
+	long int pid, uid, duration;
```

Comment:
Also wanted to mention, that currently (and previously before this commit) traced pid could be different from what is specified by user. It could happen because `strtol` returns `long int` while `env` struct is using `int` type for `pid`.

Reply:
right, we could check for valid range in user-space, but likelihood of someone specifying 123123123123123 as uid or pid and expecting it to work is very small :)

Reply:
It's small, but it still isn't nice. Also, corner cases sometimes accumulate, once I was witness like issue like this in small CLI tool, caused bash script to misbehave in very odd way. But, somebody could claim that it was bug in script (it called tool with wrong arguments).